packages/block-editor/src/components/block-list/block-list-context.native.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * WordPress dependencies
+ */
+import { createContext, useContext } from '@wordpress/element';
+
+export const DEFAULT_BLOCK_LIST_CONTEXT = {
+	scrollRef: null,
+	blocksLayouts: { current: {} },
+	updateBlocksLayouts,
+};
+
+const Context = createContext( DEFAULT_BLOCK_LIST_CONTEXT );
+const { Provider, Consumer } = Context;
+
+/**
+ * Finds a block's layout data within the provided data object.
+ *
+ * @param {Object} data     Blocks layouts list.
+ * @param {string} clientId Block's clientId.
+ *
+ * @return {Object} Found block layout data.
+ */
+function findBlockLayoutByClientId( data, clientId ) {
```

Comment:
We might consider exposing this function via the context as it would be helpful for querying the blocks layouts, wdyt?

Reply:
Cool! I updated it in https://github.com/WordPress/gutenberg/pull/39089/commits/23f2c39ffb61d4d20ecda44402807e4b7a63c6d8